Riot/Utils/UniversalLink.m

```diff
@@ -51,6 +52,13 @@ - (void)parsePathAndQueryParams
     // Remove the first empty path param string
     pathParams = [pathParams filteredArrayUsingPredicate:[NSPredicate predicateWithFormat:@"length > 0"]];
 
+    // Handle room links with aliases/identifiers
+    if ([MXTools isMatrixRoomAlias:[_url.absoluteString stringByRemovingPercentEncoding]]
+        || [MXTools isMatrixRoomIdentifier:[_url.absoluteString stringByRemovingPercentEncoding]])
+    {
+        pathParams = @[_url.absoluteString];
```

Comment:
I wonder if this is this the right place to fix this? Strictly speaking, this isn't parsing path or query params from the URL and is instead converting the entire thing into path params.

Reply:
Yeah, its quite surprising to see these here! At some stage maybe we should add an `initWithMatrixIdentifier` or similar on `UniversalLink`. That might make it a bit clearer.